### DISCUSSION TITLE
How to partially load embedded objects?

### DISCUSSION THREAD

**Discussion by mash-g** [isar/isar#839] [Help] [answered]
Hi,

I was wondering what's the correct way to store data that only needs to be loaded a little at a time.

For example we can have a Person class which contains an embedded object; List[Car] owned by the person.
Car contains a date field for when the car was manufactured.

Now let's say I load a Person from the database, but it contains 1000s of cars dating back to 1900s.
Will all cars be loaded in memory when I load the Person?
Perhaps the client only wants to see cars manufactured in the last year.
Sure I can filter on the cars, but it would have been better to not load everything into memory to begin with.

It's possible embedded objects is not for this use case.
Instead I could have a separate Car schema/table with a column for Person Id.
Then I could use the Person Id and datetime filter to load just what I need.

Which approach is more reasonable?

Thanks.

**Comment by simc** [comment, marked as answer]
Yes you should use a separate collection or links

**Reply by mash-g** [reply]
https://isar.dev/links.html#isarlink
It sounds like load() will load all of the data.
I don't see a way to load just what I need (e.g. last 5 elements ordered by datetime).

**Reply by simc** [reply]
You can use .filter()

It's probably the easiest if you just use two separate collections without links or embedded objects

**Reply by mash-g** [reply]
Okay Simon, will do, thanks a lot.
